pandas/io/tests/test_excel.py

```diff
@@ -92,110 +95,122 @@ def read_csv(self, *args, **kwds):
         return read_csv(*args, **kwds)
 
 
-class ExcelReaderTests(SharedItems, tm.TestCase):
+class ReadingTestsBase(SharedItems):
+    # This is based on ExcelWriterBase
+    #
+    # Base class for test cases to run with different Excel writers.
+    # To add a writer test, define the following:
+    # 1. A check_skip function that skips your tests if your writer isn't
+    #    installed.
+    # 2. Add a property ext, which is the file extension that your writer
+    #    writes to. (needs to start with '.' so it's a valid path)
+    # 3. Add a property engine_name, which is the name of the writer class.
+    #    For the reader this is not used for anything at the moment.
+
+    def setUp(self):
+        self.check_skip()
+        super(ReadingTestsBase, self).setUp()
+#        self.option_name = 'io.excel.%s.reader' % self.ext.strip('.')
+#        self.prev_engine = get_option(self.option_name)
+#        set_option(self.option_name, self.engine_name)
+#
+#    def tearDown(self):
+#        set_option(self.option_name, self.prev_engine)
+
     def test_parse_cols_int(self):
-        _skip_if_no_openpyxl()
-        _skip_if_no_xlrd()
 
-        suffix = ['xls', 'xlsx', 'xlsm']
-
-        for s in suffix:
-            pth = os.path.join(self.dirpath, 'test.%s' % s)
-            xls = ExcelFile(pth)
-            df = xls.parse('Sheet1', index_col=0, parse_dates=True,
-                           parse_cols=3)
-            df2 = self.read_csv(self.csv1, index_col=0, parse_dates=True)
-            df2 = df2.reindex(columns=['A', 'B', 'C'])
-            df3 = xls.parse('Sheet2', skiprows=[1], index_col=0,
-                            parse_dates=True, parse_cols=3)
-            # TODO add index to xls file)
-            tm.assert_frame_equal(df, df2, check_names=False)
-            tm.assert_frame_equal(df3, df2, check_names=False)
+        pth = os.path.join(self.dirpath, 'test1' + self.ext)
+        pref = os.path.join(self.dirpath, 'test1.csv')
+        # the truth is obtained form read_csv with Python engine
+        dfref = self.read_csv(pref, index_col=0, parse_dates=True)
```

Comment:
This block of code is repeated quite a few times in the test... can you make it a method instead?